File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        int n = A.size();
        vector<bool> presentA(n + 1, false);
        vector<bool> presentB(n + 1, false);
        vector<int> ans(n, 0);
        int cnt = 0;
        for (int i = 0; i < n; i++) {
            cnt += presentB[A[i]];
            cnt += presentA[B[i]];
            cnt += A[i] == B[i];
            presentA[A[i]] = true;
            presentB[B[i]] = true;
            ans[i] = cnt;
        }
        return ans;
    }
};

File: 2766-find-the-prefix-common-array-of-two-arrays/2766-find-the-prefix-common-array-of-two-arrays.cs
public class Solution {
    public int[] FindThePrefixCommonArray(int[] A, int[] B) {
        int n = A.Length;
        var ans = new int[n];
        
	    var presentA = new bool[n + 1];
	    var presentB = new bool [n + 1];
	    var cnt = 0;
	    for (var i = 0; i < n; i++) {
		    cnt += A[i] == B[i] ? 1 : 0;
		    cnt += presentA[B[i]] ? 1 : 0;
		    cnt += presentB[A[i]] ? 1 : 0;
		    presentA[A[i]] = true;
		    presentB[B[i]] = true;
		    ans[i] = cnt;
	    }
	    return ans;
    }
}